lib/openshift/machine_set.rb

```diff
@@ -7,16 +7,19 @@ class MachineSet < ProjectResource
 
     def desired_replicas(user: nil, cached: true, quiet: false)
       rr = raw_resource(user: user, cached: cached, quiet: quiet)
-      rr.dig('spec', 'replicas')
+      rr.dig('spec', 'replicas').to_i
     end
 
-    def current_replicas(user: nil, cached: true, quiet: false)
+    def available_replicas(user: nil, cached: true, quiet: false)
       rr = raw_resource(user: user, cached: cached, quiet: quiet)
-      rr.dig('status', 'replicas')
+      rr.dig('status', 'availableReplicas').to_i
     end
 
-    def healthy?
-      return desired_replicas == current_replicas
+    def ready?(user: nil, cached: false, quiet: false)
+      result = {}
+      result[:success] = (desired_replicas(user: user, cached: cached, quiet: quiet) ==
+        available_replicas(user: user, cached: cached, quiet: quiet))
```

Comment:
My suggestion was to use `cached: true` on the second call because you don't need to get full YAML twice. First call retrieves an updated YAML. Second call only needs to read it.

Reply:
Thanks @akostadinov. I was using `cached: true`, but as I had tested, it returns the machine_set with `desired_replicas=1` and `available_replicas=1` after I scale it to 3.  I set it to false to retrieve the data again so that it has new value.

Reply:
As explained above, then `machine_set.wait_till_ready(admin, 600)` will immediately returns  because it compares the cached `desired_replicas` with cached `available_replicas` and believes it's ready.  @akostadinov Will this method here make sense to you?

Reply:
You need `cached: false` on first call to update the cache. The second call you can have `cached: true` because cache is already updated. If you keep `cached: false` in both places you perform 2 separate calls which is unnecessary.
Maybe I was not clear the first call should be `false` and only the second one `true`.

Reply:
Thanks. Now I get it, I'll address this in a subsequent PR.